spynnaker/pyNN/models/pynn_projection_common.py

```diff
@@ -62,15 +62,26 @@ def __init__(
         post_synaptic_population._get_vertex.set_synapse_dynamics(
             synapse_dynamics_stdp)
 
-        # Set and store information for future processing
-        self._synapse_information = SynapseInformation(
-            connector, synapse_dynamics_stdp, synapse_type)
+        # Set and store synapse information for future processing
+        if hasattr(synapse_dynamics_stdp, 'weight'):
+            self._synapse_information = SynapseInformation(
+                connector, synapse_dynamics_stdp, synapse_type)
+        else:
+            self._synapse_information = SynapseInformation(
+                connector, synapse_dynamics_stdp, synapse_type,
+                connector.get_weight(), connector.get_delay())
+
```

Comment:
Rather than doing this here, I think this function should take a synapse type (not the same as the existing synapse type!), which can provide the weights and delays.  PyNN8 should provide this anyway...